cmake/config.cmake

```diff
@@ -143,6 +143,12 @@ set(USE_ROCBLAS OFF)
 # Whether use contrib sort
 set(USE_SORT ON)
 
+# Whether use contrib extern (use ";" to separate multiple externs)
+# Available externs:
+#   gcc
+#   dnnl
+set(USE_EXTERN none)
```

Comment:
Given the number of external compilers and runtimes we may have, I think it's better for each one to have its own field in config.cmake. For example, `USE_GCC` or `USE_DNNL`. With this, we can be more flexible with our options for finding the compiler / runtime. For example, `USE_GCC=ON` vs `USE_GCC=<path to gcc>`

Reply:
I think they should be prefixed as USE_EXTERNAL_<name> to make it clear this is part of the external integration if we go down that path.